lib/Controller/SAMLController.php

```diff
@@ -99,6 +99,12 @@ private function autoprovisionIfPossible(array $auth) {
 				$uid = $auth[$uidMapping];
 			}
 
+			// make sure that a valid UID is given
+			if (empty($uid)) {
+				$this->logger->error('Uid "' . $uid . '" is not a valid uid please check your attribute mapping', ['app' => $this->appName]);
+				throw new \InvalidArgumentException('No valid uid given, please check your attribute mapping. Given uid: ' . $uid);
```

Comment:
Throw `NoUserFoundException` instead, so it is caught in the same way,
if not update the docs as well

Reply:
I chosed the "InvalidArgumentException" because it is used here as well if no valid user is given: https://github.com/nextcloud/user_saml/pull/174/files#diff-08ab8d6da860d4cc7a5530b6414b0f6eR238 But beside that I don't care that much

Reply:
The calles of this method wrap it into a try block and only catch NoUserFoundException however.

Reply:
yes, because for this exception we have a dedicated error page. For the other case not. As in this case the useful information are in the log I think the generic error page which points you to the log makes much more sense than a "user not provisioned" error page because that's not the problem here.

Reply:
OK